flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamEdge.java

```diff
@@ -48,15 +49,25 @@
 	 * output selection).
 	 */
 	private final List<String> selectedNames;
+
+	/**
+	 * The side-output tag (if any) of this {@link StreamEdge}.
+	 */
+	private final OutputTag outputTag;
+
+	/**
+	 * The {@link StreamPartitioner} on this {@link StreamEdge}.
+	 */
 	private StreamPartitioner<?> outputPartitioner;
 
 	public StreamEdge(StreamNode sourceVertex, StreamNode targetVertex, int typeNumber,
-			List<String> selectedNames, StreamPartitioner<?> outputPartitioner) {
+			List<String> selectedNames, StreamPartitioner<?> outputPartitioner, OutputTag outputTag) {
 		this.sourceVertex = sourceVertex;
 		this.targetVertex = targetVertex;
 		this.typeNumber = typeNumber;
 		this.selectedNames = selectedNames;
 		this.outputPartitioner = outputPartitioner;
+		this.outputTag = outputTag;
 
```

Comment:
Not sure what the edge id exactly does and who uses it so I prefer to not touch it, for now.